### ISSUE TITLE
touchfile does not update modification time

### ISSUE THREAD

**Issue by sol** [yesodweb/Shelly.hs#102]
(this is at least the true on Linux x64)

**Comment by gregwebs** [comment]
Right now it is using

    Data.Text.IO.appendFile file ""

I could see how something along the way could end up optimizing this to a no-op. Better would be to use a real touch command, do you know what that is?

**Comment by sol** [comment]
Not efficient, but good enough for testing purpose: https://github.com/hspec/mockery/pull/4/files

Also, @Fuuzetsu mentioned there will be a way to set modification times with the next version of `directory`.

**Comment by gregwebs** [comment]
> Not efficient, but good enough for testing purpose: https://github.com/hspec/mockery/pull/4/files

yikes! we can't do that!

> Also, @Fuuzetsu mentioned there will be a way to set modification times with the next version of directory

Shelly can upgrade to it, but it will need to use CPP to continue to support older versions of directory.

**Comment by damiencourousse** [comment]
I have just stumbled on this issue as well, also on Linux x64.
Any update?

**Comment by gregwebs** [comment]
@damiencourousse do you know a good API for touching a file in Haskell?

**Comment by damiencourousse** [comment]
> @damiencourousse do you know a good API for touching a file in Haskell?

No, I'm sorry.   I have seen that Turtle uses `touchFile` from module `System.Posix` for all OSes but Windows:
https://hackage.haskell.org/package/turtle-1.2.5/docs/src/Turtle-Prelude.html#touch

**Comment by andreasabel** [comment]
Does not work on Mac OS X either.  
Seems to be known on stackoverflow: https://stackoverflow.com/a/50298141/1817682.
(I am actually looking for an OS-oblivious version of `touch` in Haskell.  Maybe if I skip Win32, the previous answer is sufficient.)
This issue could maybe be fixed by stealing the Turtle code.

**Comment by gregwebs** [comment]
want to send a pull request to fix it?

**Comment by andreasabel** [comment]
That would probably not be too hard.
But I am afraid of breaking things; the CI isn't up to speed: I reported #191.
Getting the CI running again is maybe the highest priority.  After that, folks could more confidently submit PRs. Friendly contributors have set this up Github CI Actions for Agda: https://github.com/agda/agda/actions. This test compilation on version 8.0-8.10 of GHC and on the three major OSs.
